hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/state/ArtifactUploadState.java

```diff
@@ -197,4 +178,12 @@ public void setNoDataAvilableSoftwareModule(final boolean noDataAvilableSoftware
         this.noDataAvilableSoftwareModule = noDataAvilableSoftwareModule;
     }
 
+    public Optional<SoftwareModule> getSelectedBaseSoftwareModule() {
+        return selectedBaseSoftwareModule == null ? Optional.empty() : Optional.of(selectedBaseSoftwareModule);
```

Comment:
The if-statement is superfluous. Consider the use of _Optional.ofNullable(...)_.
(see http://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ofNullable-T-)